pkg/cloudprovider/vsphereparavirtual/loadbalancer_test.go

```diff
@@ -44,385 +40,310 @@ import (
+			assert.Equal(t, vmservice.ErrVMServiceIPNotFound, err)
+
+			// Update the service definition to add ports
+			testK8sService.Spec = v1.ServiceSpec{
+				Ports: []v1.ServicePort{
+					{
+						Name:     "test-port",
+						Port:     80,
+						NodePort: 30900,
+						Protocol: "TCP",
 					},
-				}
+				},
+			}
 
+			if testCase.expectErr {
 				// Ensure that the client Update call returns an error on update
 				fcw.UpdateFunc = func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
 					return fmt.Errorf("Some undefined update error")
 				}
-			})
-			It("should return error", func() {
-				Expect(err).Should(HaveOccurred())
-			})
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
 		})
+	}
+}
 
-		Context("when VMService is updated", func() {
-			BeforeEach(func() {
-				testK8sService = &v1.Service{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      testK8sServiceName,
-						Namespace: testK8sServiceNameSpace,
-					},
-				}
-
-				// Add the service with no ports
-				_, err := lb.EnsureLoadBalancer(context.Background(), testClustername, testK8sService, []*v1.Node{})
-				Expect(err).Should(Equal(vmservice.ErrVMServiceIPNotFound))
-
-				// Update the service definition to add ports
-				testK8sService.Spec = v1.ServiceSpec{
-					Ports: []v1.ServicePort{
+func TestEnsureLoadBalancer_VMServiceExternalTrafficPolicyLocal(t *testing.T) {
+	lb, fcw := newTestLoadBalancer()
+	testK8sService := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testK8sServiceName,
+			Namespace: testK8sServiceNameSpace,
+		},
+		Spec: v1.ServiceSpec{
+			ExternalTrafficPolicy: v1.ServiceExternalTrafficPolicyTypeLocal,
+		},
+	}
+	fcw.CreateFunc = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+		vms := &vmopv1alpha1.VirtualMachineService{
+			Status: vmopv1alpha1.VirtualMachineServiceStatus{
+				LoadBalancer: vmopv1alpha1.LoadBalancerStatus{
+					Ingress: []v1alpha1.LoadBalancerIngress{
 						{
-							Name:     "test-port",
-							Port:     80,
-							NodePort: 30900,
-							Protocol: "TCP",
+							IP: "10.10.10.10",
 						},
 					},
-				}
-			})
-			It("should not return error", func() {
-				Expect(err).ShouldNot(HaveOccurred())
-			})
-		})
-	})
+				},
+			},
+		}
+		vms.DeepCopyInto(obj.(*vmopv1alpha1.VirtualMachineService))
+		return nil
+	}
 
-	Describe("EnsureLoadBalancer", func() {
-		var (
-			status    *v1.LoadBalancerStatus
-			ensureErr error
-		)
+	_, ensureErr := lb.EnsureLoadBalancer(context.Background(), testClustername, testK8sService, []*v1.Node{})
+	assert.NoError(t, ensureErr)
 
-		JustBeforeEach(func() {
-			status, ensureErr = lb.EnsureLoadBalancer(context.Background(), testClustername, testK8sService, []*v1.Node{})
-		})
+	err := lb.EnsureLoadBalancerDeleted(context.Background(), testClustername, testK8sService)
+	assert.NoError(t, err)
+}
+
+func TestEnsureLoadBalancer_VMServiceCreatedIPNotFound(t *testing.T) {
```

Comment:
nit: These two ensure could be refactored to use testcases slice with variable called `createFunc`